### PR TITLE
add Read, Update, List handlers

### PR DESCRIPTION
**Pull request by sfbaker7**
*Issue #, if available:*

*Description of changes:*

Adds handlers for each of Read, Update, List operations

See which event is invoked when: https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-develop.html

*Testing*

Integ Tests + Unit Tests

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running software.amazon.logs.querydefinitions.ReadHandlerTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 4.024 s - in software.amazon.logs.querydefinitions.ReadHandlerTest
[INFO] Running software.amazon.logs.querydefinitions.CreateHandlerTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.091 s - in software.amazon.logs.querydefinitions.CreateHandlerTest
[INFO] Running software.amazon.logs.querydefinitions.ListHandlerTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.031 s - in software.amazon.logs.querydefinitions.ListHandlerTest
[INFO] Running software.amazon.logs.querydefinitions.UpdateHandlerTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.056 s - in software.amazon.logs.querydefinitions.UpdateHandlerTest
[INFO] Running software.amazon.logs.querydefinitions.DeleteHandlerTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.059 s - in software.amazon.logs.querydefinitions.DeleteHandlerTest
[INFO]
[INFO] Results:
[INFO]
[INFO] Tests run: 18, Failures: 0, Errors: 0, Skipped: 0
[INFO]
```

### CODE REVIEW

File: aws-logs-querydefinitions/src/main/java/software/amazon/logs/querydefinitions/ReadHandler.java [hunk 1]
```diff
@@ -17,11 +27,29 @@
...
+                if (queryDefinition.isPresent()) {
+                    return ProgressEvent.<ResourceModel, CallbackContext>builder()
+                            .resourceModel(model)
+                            .status(OperationStatus.SUCCESS)
+                            .build();
+                }
+            } catch (InvalidParameterException ex) {
+                throw new CfnInvalidRequestException(ResourceModel.TYPE_NAME, ex);
+            } catch (ServiceUnavailableException ex) {
+                throw new CfnServiceInternalErrorException(ResourceModel.TYPE_NAME, ex);
```
Comment (anugarg07) [inline aws-logs-querydefinitions/src/main/java/software/amazon/logs/querydefinitions/ReadHandler.java:49]: So our `DescribeQueryDefinition` API does not throw any error if resource is not found ?

File: aws-logs-querydefinitions/src/test/java/software/amazon/logs/querydefinitions/ListHandlerTest.java [hunk 2]
```diff
@@ -27,14 +40,17 @@
     public void setup() {
         proxy = mock(AmazonWebServicesClientProxy.class);
         logger = mock(Logger.class);
+        handler = new ListHandler();
     }
 
     @Test
     public void handleRequest_SimpleSuccess() {
```
Comment (anugarg07) [inline aws-logs-querydefinitions/src/test/java/software/amazon/logs/querydefinitions/ListHandlerTest.java:47]: Can we add case when nextToken != null and we make multiple calls to CWL ?

**Review by anugarg07 — COMMENTED** [review]
Looks good overall ; some comments

File: aws-logs-querydefinitions/src/main/java/software/amazon/logs/querydefinitions/ReadHandler.java (hunk 1 above)
Comment (sfbaker7) [reply aws-logs-querydefinitions/src/main/java/software/amazon/logs/querydefinitions/ReadHandler.java:49]: correct, it does not

File: aws-logs-querydefinitions/src/test/java/software/amazon/logs/querydefinitions/ListHandlerTest.java (hunk 2 above)
Comment (sfbaker7) [reply aws-logs-querydefinitions/src/test/java/software/amazon/logs/querydefinitions/ListHandlerTest.java:47]: yes

File: aws-logs-querydefinitions/src/main/java/software/amazon/logs/querydefinitions/ReadHandler.java (hunk 1 above)
Comment (anugarg07) [reply aws-logs-querydefinitions/src/main/java/software/amazon/logs/querydefinitions/ReadHandler.java:49]: okay

**Pull request merged**